src/Backend/JSON.idr

```diff
@@ -27,47 +31,55 @@ nary name = map ((name ++) . show . finToNat) range
 defRef : Name -> JSON
 defRef name = JObject [("$ref", JString $ "#/definitions/" ++ name)]
 
+JSONLookupM : Type -> Type
+JSONLookupM t = LookupM JSON t
+
+JSONMakeDefM : Type -> Type
+JSONMakeDefM t = MakeDefM JSON t
+
 makeSubSchema' : TNamed 0 -> JSON
 makeSubSchema' = defRef . name
 
 mutual
-  makeSubSchema : TDef 0 -> JSON
-  makeSubSchema  T0         = defRef "emptyType"
-  makeSubSchema  T1         = defRef "singletonType"
+  makeSubSchema : TDef 0 -> JSONLookupM JSON
+  makeSubSchema  T0         = pure $ defRef "emptyType"
+  makeSubSchema  T1         = pure $ defRef "singletonType"
   makeSubSchema (TSum ts)   = disjointSubSchema (zip (nary "case") ts)
   makeSubSchema (TProd ts)  = productSubSchema (nary "proj") ts
-  makeSubSchema (TMu cs)    = defRef (nameMu cs)
-  makeSubSchema (TApp f []) = defRef . name $ f
-  makeSubSchema (TApp f xs) = defRef . name $ f `apN` xs
+  makeSubSchema (TMu cs)    = pure $ defRef (nameMu cs)
+  makeSubSchema (TApp f []) = pure $ defRef . name $ f
+  makeSubSchema (TApp f xs) = pure $ defRef . name $ f `apN` xs
+  makeSubSchema (TRef n)    = raise $ RefNotFound n
   
   ||| Generate a schema that matches exactly one of the supplied schemas, which must be wrapped in its corresponding name.
-  disjointSubSchema : Vect k (Name, TDef 0) -> JSON
-  disjointSubSchema cases = JObject [("oneOf", JArray . toList $ map makeCase cases)]
+  disjointSubSchema : Vect k (Name, TDef 0) -> JSONLookupM JSON
+  disjointSubSchema cases = 
+    pure $ JObject [("oneOf", JArray . toList $ !(traverseEffect (assert_total makeCase) cases))]
     where
     isMu : TDef n -> Bool
     isMu (TMu _) = True
     isMu _       = False
 
-    makeCase : (Name, TDef 0) -> JSON
-    makeCase (name, td) = JObject
+    makeCase : (Name, TDef 0) -> JSONLookupM JSON
+    makeCase (name, td) = pure $ JObject
```

Comment:
```suggestion
    makeCase (name, td) = pure JObject
```

Reply:
I reverted the change because it didn't compile